app/src/main/java/com/example/calculator/view/MainActivity.java

```diff
@@ -2,31 +2,39 @@
 
 import android.os.Bundle;
 import android.view.View;
-import android.widget.TextView;
+import android.widget.CompoundButton;
+import android.widget.ToggleButton;
 
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.ViewModelProvider;
 
 import com.example.calculator.R;
 import com.example.calculator.databinding.ActivityMainBinding;
+import com.example.calculator.viewmodel.MainViewModel;
 
 
 public class MainActivity extends AppCompatActivity implements View.OnClickListener {
 
     private ActivityMainBinding binding;
-    private TextView current;
+
+    private MainViewModel viewModel;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        String TAG = "zero to appear";
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
         binding = ActivityMainBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
         binding.display.setText(R.string.zero);
```

Comment:
You should try hint in the xml rather than setting this to 0. if you do want to set it to 0, you should do that in the xml as the `android:text`